src/Skybrud.Umbraco.Redirects/Models/RedirectsRepository.cs

```diff
@@ -308,25 +308,19 @@ public class RedirectsRepository {
             // Some input validation
             if (String.IsNullOrWhiteSpace(url)) throw new ArgumentNullException(nameof(url));
 
-			var fullUrl = url + (queryString.IsNullOrWhiteSpace() ? "" : "?" + queryString);
-
-			url = url.TrimEnd('/').Trim();
-            queryString = (queryString ?? "").Trim();
-
             // Just return "null" if the table doesn't exist (since there aren't any redirects anyway)
             if (!SchemaHelper.TableExist(RedirectItemRow.TableName)) return null;
 
-            // Fetch non-regex redirects from the database
-            Sql sql = new Sql().Select("*").From(RedirectItemRow.TableName).Where<RedirectItemRow>(x => !x.IsRegex && x.Url == url && x.QueryString == queryString || x.ForwardQueryString);
-            List<RedirectItem> redirects = Database.Fetch<RedirectItemRow>(sql).Select(RedirectItem.GetFromRow).ToList();
+			url = url.TrimEnd('/').Trim();
+			queryString = (queryString ?? string.Empty).Trim();
+			var fullUrl = url + (queryString.IsNullOrWhiteSpace() ? string.Empty : '?' + queryString);
 
-            // Fetch regex redirects from the database
-            sql = new Sql().Select("*").From(RedirectItemRow.TableName).Where<RedirectItemRow>(x => x.IsRegex);
-            redirects.AddRange(Database.Fetch<RedirectItemRow>(sql).Where(x => Regex.IsMatch(fullUrl, x.Url)).Select(RedirectItem.GetFromRow));
+			// Fetch redirects from the database and filter regex
+			Sql sql = new Sql().Select("*").From(RedirectItemRow.TableName).Where<RedirectItemRow>(x => (!x.IsRegex && x.Url == url && x.QueryString == queryString) || x.IsRegex);
+            List<RedirectItem> redirects = Database.Fetch<RedirectItemRow>(sql).Where(x => !x.IsRegex || Regex.IsMatch(fullUrl, x.Url)).Select(RedirectItem.GetFromRow).ToList();
 
             // Return a combined list of the redirects
             return redirects.OrderBy(x => x.RootNodeId > 0 ? "0" : "1").ToArray();
```

Comment:
Why are the redirects sorted on `RootNodeId`? Other methods don't do this, so not sure if this can also be simplified...

Reply:
The idea is that you can create redirects that are either at a global level or for a specific site/domain. The latter type is more specific than a global redirect, and therefore the matched redirects are sorted with non-global redirects first. 

As the `GetRedirectsByUrl` method may return more than one, it sorts the redirects by whether the root ID is above `0` or not.

The `GetRedirectByUrl` methods only return a single redirect, and also takes the `rootNodeId` as a parameter along with the URL, so the same logic isn't applied here. Query string forwarding does however create a bit of ambiguity on what redirect is returned, as more redirects may match the same input URL.